### PR TITLE
docs: Add some docs on plugins

### LINKED ISSUE #4243: Missing documentation for plugins
#### Description:

The changelog for 0.4.11-rc1 mentions the new plugin system and the git/ethereum plugins. I can't find any documentation on how to use them though. Also, is the plugin system experimental?

### PR DESCRIPTION
**Pull request by magik6k**
Closes #4243

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Plugins
+
+Since 0.4.11 go-ipfs has a plugin system for various added functionality.
+
+When IPFS node is created, it will load plugins from `.ipfs/plugins` directory.
+
+### Plugin types
+
+#### IPLD
+IPLD plugins add support for additional LD formats to `ipfs dag` and other IPLD
```
Comment (Stebalien) [inline docs/plugins.md]: `s/additional LD/additional IPLD/`. We don't support other LD formats (JSON-LD etc...).

File: docs/plugins.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Plugins
+
+Since 0.4.11 go-ipfs has a plugin system for various added functionality.
+
+When IPFS node is created, it will load plugins from `.ipfs/plugins` directory.
```
Comment (Stebalien) [inline docs/plugins.md]: To be precise, this should probably say "will load plugins from `$IPFS_PATH/plugins` (where `$IPFS_PATH` defaults to `~/.ipfs`)" or something like that.

File: docs/plugins.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Plugins
+
+Since 0.4.11 go-ipfs has a plugin system for various added functionality.
```
Comment (Stebalien) [inline docs/plugins.md]: Should also note that this only works on Linux.

Comment (Stebalien) [reply docs/plugins.md]: Also, we should explicitly state that this is "experimental".

**Comment by magik6k** [comment]
Updated with section on preload

**Comment by lgierth** [comment]
Cool thanks, this looks good! Could you also:

- [ ] add it to experimental-features.md
- [ ] link it in the plugins section in the v0.4.11 changlog

**Comment by magik6k** [comment]
LGTM

**Pull request merged**